test/stability/testMerge/testMerge.go

```diff
@@ -89,151 +121,350 @@ func main() {
 			return err
 		}
-
-		// Start pipeline
-		go func() {
-			err = flow.SystemStart()
-		}()
+		checkedFlow, err = setGetter(recv, inport1)
 		if err != nil {
 			return err
 		}
-		progStart = time.Now()
+	}
+	if testScenario != receivePart {
+		if err := flow.SetHandler(checkedFlow, checkPackets, nil); err != nil {
+			return err
+		}
+		if err := flow.SetStopper(checkedFlow); err != nil {
+			return err
+		}
+	} else {
+		if err := flow.SetHandler(checkedFlow, fixPackets, nil); err != nil {
+			return err
+		}
+		if err := flow.SetSender(checkedFlow, uint16(outport1)); err != nil {
+			return err
+		}
+	}
+	if testScenario == receivePart {
+		return flow.SystemStart()
+	}
+	// Start pipeline
+	go func() {
+		err = flow.SystemStart()
+	}()
+	if err != nil {
+		return err
+	}
+	progStart = time.Now()
 
-		// Wait for enough packets to arrive
-		testDoneEvent.L.Lock()
-		testDoneEvent.Wait()
-		testDoneEvent.L.Unlock()
+	// Wait for enough packets to arrive
+	testDoneEvent.L.Lock()
+	testDoneEvent.Wait()
+	testDoneEvent.L.Unlock()
 
-		return composeStatistics()
+	return composeStatistics(scenario)
+}
+
+// stop system and reset test counters
+func resetState() {
+	flow.SystemStop()
+
+	sentPacketsGroup1 = 0
+	sentPacketsGroup2 = 0
+
+	recvPacketsGroup1 = 0
+	recvPacketsGroup2 = 0
+	recvCopyPacketsGroup1 = 0
+	recvCopyPacketsGroup2 = 0
+
+	recvPackets = 0
+	brokenPackets = 0
+
+	passed = 1
+}
+
+// Init NFF-GO system
+func initDPDK() error {
+	config := flow.Config{
+		DPDKArgs: []string{dpdkLogLevel},
+	}
+	return flow.SystemInit(&config)
+}
+
+func executeTest(configFile, target string, testScenario uint, scenario pipelineScenario, generatorType getScenario, addSegmAfterMerge bool) error {
+	if testScenario > 3 || testScenario < 0 {
+		return errors.New("testScenario should be in interval [0, 3]")
 	}
-	return nil
+	stabilityCommon.InitCommonState(configFile, target)
+
+	fixMACAddrs1 = stabilityCommon.ModifyPacket[0].(func(*packet.Packet, flow.UserContext))
+	fixMACAddrs2 = stabilityCommon.ModifyPacket[1].(func(*packet.Packet, flow.UserContext))
+	fixMACAddrs = stabilityCommon.ModifyPacket[0].(func(*packet.Packet, flow.UserContext))
+	return setTestingPart(scenario, generatorType, addSegmAfterMerge)
 }
 
-func composeStatistics() error {
-	// Compose statistics
-	sent1 := atomic.LoadUint64(&sentPacketsGroup1)
-	sent2 := atomic.LoadUint64(&sentPacketsGroup2)
-	sent := sent1 + sent2
+func failCheck(msg string) error {
+	println(msg)
+	println("TEST FAILED")
+	return errors.New(msg)
+}
+
+func composeStatistics(scenario pipelineScenario) error {
+	maxBroken := totalPackets / 10
+	passedLimit := uint64(98)
+	lessPercent := 50
+	eps := 2
+	var sent1, sent2, shouldGet1, shouldGet2, shouldGetCopy1, shouldGetCopy2 uint64
+	sent1 = atomic.LoadUint64(&sentPacketsGroup1)
+	sent2 = atomic.LoadUint64(&sentPacketsGroup2)
+	shouldGet1 = sent1
+	shouldGet2 = sent2
+	switch scenario {
+	case singleSegm:
+		shouldGet2 = 0
+		lessPercent = 100
+		eps = 0
```

Comment:
Really? Looks like it will be always fluky

Reply:
No, we send only one group of packets, thats why it will always be 100: 0 percent group ratios

Reply:
We check send\recv total ratio, this is a group ratio.